test/test_main.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+from binascii import unhexlify
+from pathlib import Path
+from sys import version_info
+from warnings import filterwarnings
+
+import pytest
+
+from puremagic.main import what
+
+filterwarnings("ignore", message="'imghdr' is deprecated")
+try:  # imghdr was removed from the standard library in Python 3.13
+    from imghdr import what as imghdr_what
+except ModuleNotFoundError:
+    imghdr_what = None  # type: ignore[assignment]
+
+# file_tests = sorted(test_func.__name__[5:] for test_func in imghdr.tests)
+# file_tests = "bmp exr gif jpg pbm pgm png ppm ras rgb tif webp xbm".split()
+file_tests = "bmp gif jpg png tif webp".split()  # TODO: (cclauss) Add the remaining files
+
+
+@pytest.mark.skipif(imghdr_what is None, reason="imghdr was removed from the standard library in Python 3.13")
+@pytest.mark.parametrize("file", file_tests)
+def test_what_from_file(file, h=None):
+    """Run each test with a path string and a pathlib.Path."""
+    file = f"test/resources/images/test.{file}"
+    assert what(file, h) == imghdr_what(file, h)
+    file = Path(file).resolve()
+    assert what(file, h) == imghdr_what(file, h)
+
+
+@pytest.mark.skipif(imghdr_what is None, reason="imghdr was removed from the standard library in Python 3.13")
+def test_what_from_file_none(file="test/resources/fake_file", h=None):
+    assert what(file, h) == imghdr_what(file, h) is None
+    file = Path(file).resolve()
+    assert what(file, h) == imghdr_what(file, h) is None
+
+
+string_tests = [
```

Comment:
I would prefer if we can generate as many of these as real files (using `test.bmp` as base converted from) over time, but very happy for now they have at least been included, thank you!